tensorflow/python/keras/saving/pickle.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Support for Python's Pickle protocol by implementing
+packing/unpacking of Keras Models into objects which are themselves
+picklable.
+"""
+
+import os
+import tarfile
+
+from io import BytesIO
+from uuid import uuid4
+
+import numpy as np
+
+from tensorflow.python.platform import gfile
+from tensorflow.python.keras.models import load_model
+
+
+
+def unpack_model(packed_keras_model):
+  """Reconstruct a Model from the result of pack_model.
+
+  Args:
+      packed_keras_model (np.array): return value from pack_model.
+
+  Returns:
+      tf.keras.Model: a Keras Model instance.
+  """
+  temp_dir = f"ram://{uuid4()}"
+  b = BytesIO(packed_keras_model)
+  with tarfile.open(fileobj=b, mode="r") as archive:
+    for fname in archive.getnames():
+      dest = os.path.join(temp_dir, fname)
+      gfile.makedirs(os.path.dirname(dest))
+      with gfile.GFile(dest, "wb") as f:
+        f.write(archive.extractfile(fname).read())
+  model = load_model(temp_dir)
+  # delete file by file because gfile.rmtree is not implemented for ram://
+  for _, _, filenames in gfile.walk(temp_dir):
+    for filename in filenames:
+      # Currently, gfile.walk returns
+      # the entire path for the ram:// filesystem
+      # this is inconsistent with the behavior of
+      # os.walk and gfile.walk on non-ram
+      # filesystems
```

Comment:
@mihaimaruseac would you be able to give any input on the issues described in these comments? Thanks

Reply:
This looks like a bug in the RAM filesystem implementation, if I understand the comment correctly. Can you file a separate bug for this please?

Reply:
Sure, I just wanted to confirm that it is a bug, happy to file an issue, I'll tag you

Reply:
Opened #47313 and #47316